zookeeper-server/src/main/java/org/apache/zookeeper/server/persistence/TxnLogToolkit.java

```diff
@@ -361,6 +371,40 @@ static String getFormattedTxnStr(Record txn) {
         return txnData.toString();
     }
 
+    private static Record deserializeTxn(Txn txn) throws IOException {
+        ByteBuffer bb = ByteBuffer.wrap(txn.getData());
+        Record record;
+        switch (txn.getType()) {
+            case ZooDefs.OpCode.create:
+            case ZooDefs.OpCode.create2:
+                record = new CreateTxn();
+                break;
+            case ZooDefs.OpCode.createTTL:
+                record = new CreateTTLTxn();
+                break;
+            case ZooDefs.OpCode.createContainer:
+                record = new CreateContainerTxn();
+                break;
+            case ZooDefs.OpCode.delete:
+            case ZooDefs.OpCode.deleteContainer:
+                record = new DeleteTxn();
+                break;
+            case ZooDefs.OpCode.setData:
+                record = new SetDataTxn();
+                break;
+            case ZooDefs.OpCode.error:
+                record = new ErrorTxn();
+                break;
+            case ZooDefs.OpCode.check:
+                record = new CheckVersionTxn();
+                break;
+            default:
+                return null;
```

Comment:
We should also throw exception here, just like `SerializeUtils.deserializeTxn`. Silence is not good in dumping.